src/services/event.js

```diff
@@ -5,98 +5,124 @@
  */
 import { DateTime } from 'luxon';
 import http from './http';
+import session from './session';
 
 /**
  * The first element is the start time of the event, the second element is the end time of the
  * event, and the third element is the location of the event.
  * @global
- * @typedef {Array} EventOccurrence
- * @example
- * [
- *   "2017-11-18T19:00:00-05:00",
- *   "2017-11-18T22:00:00-05:00",
- *   "Tavilla Hall 126 - Conference Room"
- * ]
+ * @typedef EventOccurrence
+ * @property {string} StartDate The datetime, as a string, when the occurence begins
+ * @property {string} EndDate The datetime, as a string, when the occurence ends
+ * @property {string} Location The location of the occurence
  */
 
 /**
  * @global
  * @typedef Event
- * @property {String} Event_ID
- * @property {String} Event_Name
- * @property {String} Event_Title
- * @property {String} Event_Type_Name
- * @property {Number} Category_Id
- * @property {String} Description
- * @property {EventOccurrence[]} Occurrences
+ * @property {String} Event_ID The ID of the event, from 25Live
+ * @property {String} Event_Name The internal name of the event
+ * @property {String} Event_Title The title, to be displayed, of the event
+ * @property {boolean} HasCLAWCredit Whether the event offers CL&W Credit
+ * @property {String} Description The description of the event
+ * @property {EventOccurrence[]} Occurrences All scheduled occurrences of the event
  */
 
 /**
- * Get all events
- * @return {Promise.<Event[]>} returns all the events
+ * @global
+ * @typedef AttendedEvent
+ * @property {String} CHDate The date and time that the user received CL&W credit
+ * @property {String} CHTermCD Term code of the event
+ * @property {String} Description Given description of the event
+ * @property {String} Event_Name The generic name of the event
+ * @property {String} Event_Title Specific title of the event
+ * @property {EventOccurrence[]} Occurrences All scheduled occurrences of the event
+ * @property {String} Organization Organization hosting the event
+ * @property {Number} Required Required CL&W credits for the user
  */
 
-const getAllEvents = () => http.get('events/25Live/All');
+/**
+ * Gets all events from the backend, and then formats and sorts them
+ * @returns {Event[]} All events
+ */
+const getAllEvents = async () => {
+  const allEvents = await http.get('events/25Live/All');
+  return allEvents.map((e) => formatevent(e)).sort(sortEventsByTime);
+};
 
-const getAllCLAWEvents = () => http.get('/events/25Live/CLAW');
+/**
+ * Gets upcoming CL&W events and formats them for display
+ * TODO: Unused. Consider removing
```

Comment:
You would think so, but no it's not. Currently, the CLW filter is applied to the events that have already been fetched (e.g. by getAllEvents), using the HasCLAWCredit attribute (which is functionally the same as filtering for which of the events have category 85, but simpler because that work has already been done by the backend).

The question of whether filtering should be done in the backend or the front end is tricky. Usually, you want those kinds of resource intensive operations done by the backend so that the user's experience is fluid and quick. However, achieving the kind of granular filtering we have designed via a REST API is quite tricky and would result in a many many requests back and forth. For now, I've chosen to continue the already established pattern of filtering on the front end, but we might revisit this eventually.